lib/symboldatabase.cpp

```diff
@@ -5246,17 +5246,18 @@ const Enumerator * SymbolDatabase::findEnumerator(const Token * tok, std::set<st
         if (tok1->strAt(-1) == "::")
             scope = &scopeList.front();
         else {
-            // FIXME search base class here
-
             const Scope* temp = nullptr;
             if (scope)
                 temp = scope->findRecordInNestedList(tok1->str());
             // find first scope
             while (scope && scope->nestedIn) {
                 if (!temp)
                     temp = scope->nestedIn->findRecordInNestedList(tok1->str());
-                if (!temp && scope->functionOf)
+                if (!temp && scope->functionOf) {
                     temp = scope->functionOf->findRecordInNestedList(tok1->str());
+                    if (!temp && scope->functionOf->nestedIn)
```

Comment:
Does this need to be added upon the deeper the enum is placed? If yes this should probably be a loop.

Reply:
Do you have an example? I'd rather not add something without a test.

Reply:
Just wrap both of them within another struct and/or namespace and see if it is found.

Reply:
~~~
namespace N {
    struct S1 {
        enum E { E1 } e;
    };
	namespace O {
		struct S2 {
			static void f();
		};
	}
}
void N::O::S2::f() {
    S1 s;
    s.e = S1::E1;
}
~~~
~~~
if (!temp && scope->functionOf) {
                    temp = scope->functionOf->findRecordInNestedList(tok1->str());
                    const Scope* nested = scope->functionOf->nestedIn;
                    while (!temp && nested) {
                        temp = nested->findRecordInNestedList(tok1->str());
                        nested = nested->nestedIn;
                    }
                }
~~~

Reply:
That's exactly what I was thinking of.